pkg/instancetype/compatibility/compatibility.go

```diff
@@ -0,0 +1,196 @@
+//nolint:dupl,lll,gocyclo
+package compatibility
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/json"
+
+	"kubevirt.io/api/instancetype/v1alpha1"
+	"kubevirt.io/api/instancetype/v1alpha2"
+	"kubevirt.io/api/instancetype/v1beta1"
+	generatedscheme "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/scheme"
+)
+
+func GetInstancetypeSpec(revision *appsv1.ControllerRevision) (*v1beta1.VirtualMachineInstancetypeSpec, error) {
+	if err := Decode(revision); err != nil {
+		return nil, err
+	}
+	switch obj := revision.Data.Object.(type) {
+	case *v1beta1.VirtualMachineInstancetype:
+		return &obj.Spec, nil
+	case *v1beta1.VirtualMachineClusterInstancetype:
+		return &obj.Spec, nil
+	default:
+		return nil, fmt.Errorf("unexpected type in ControllerRevision: %T", obj)
+	}
```

Comment:
```suggestion
	}
	return nil, fmt.Errorf("unexpected type in ControllerRevision: %T", obj)
```

Reply:
nit: not sure what is better

Reply:
That isn't possible due to the switch using:
```
switch obj := revision.Data.Object.(type) {
[..]
default:
	return nil, fmt.Errorf("unexpected type in ControllerRevision: %T", obj)
}
```

Reply:
ACK